src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalInstantiationCheck.java

```diff
@@ -336,9 +313,8 @@ private boolean isStandardClass(String className, String illegal) {
             // not to java.lang.Boolean
 
             final boolean isSameFile = classNames.contains(className);
-            final boolean isSamePackage = isSamePackage(className);
```

Comment:
This is removed because the result doesn't matter. With this line removed, the `isSamePackage` would pick it up at line 236. Both are almost duplicating the same logic in different ways. The only way this could possible stay is if the class we are looking for is in a different package, but if this was the case, we would need an import statement. This is all about if an import isn't given.

This also removes the deprecated class loading.